router/router.go

```diff
@@ -524,6 +543,75 @@ func (self *Router) RegisterXWebHandlerFactory(x xweb.ApiHandlerFactory) error {
 	return self.xwebFactoryRegistry.Add(x)
 }
 
+func (self *Router) initializeCtrlEndpoints() error {
+	if self.config.Ctrl.DataDir == "" {
+		return errors.New("ctrl DataDir not configured")
+	}
+	endpointsFile := path.Join(self.config.Ctrl.DataDir, "endpoints")
+	_, err := os.Stat(endpointsFile)
+	if errors.Is(err, fs.ErrNotExist) {
+		for _, ep := range self.config.Ctrl.InitialEndpoints {
+			self.ctrlEndpoints.Set(ep.String(), ep)
+		}
+		data, err := self.ctrlEndpoints.MarshalYAML()
+		if err != nil {
+			return err
+		}
+		return os.WriteFile(endpointsFile, data.([]byte), 0600)
+	}
+
+	b, err := os.ReadFile(endpointsFile)
+	if err != nil {
+		return nil
+	}
+
+	if err := yaml.Unmarshal(b, &self.ctrlEndpoints); err != nil {
+		return err
+	}
+	//TODO: Handle mismatches
+	return nil
+}
+
+func (self *Router) UpdateCtrlEndpoints(endpoints []string) error {
+	pfxlog.Logger().Info("Recieved endpoints to update:")
+	pfxlog.Logger().Info(endpoints)
+	save := false
+	newEps := make(map[string]bool)
+	for _, ep := range endpoints {
+		newEps[ep] = true
+		if !self.ctrlEndpoints.Has(ep) {
+			pfxlog.Logger().Info("Adding new endpoint")
+			save = true
+			parsed, err := transport.ParseAddress(ep)
+			if err != nil {
+				return err
+			}
+			self.ctrlEndpoints.Set(ep, NewUpdatableAddress(parsed))
+		}
+	}
+
```

Comment:
Should probably always do this check, just in case we added an endpoint and removed one at the same time, so the lengths are the same, but the contents are different